app/src/main/java/com/losandroides/learn/data/local/datasource/DefaultLocalItemDatasource.kt

```diff
@@ -0,0 +1,35 @@
+package com.losandroides.learn.data.local.datasource
+
+import arrow.core.Either
+import com.losandroides.learn.data.local.database.ItemDatabase
+import com.losandroides.learn.domain.model.Item
+import com.losandroides.learn.domain.toDatabaseModel
+import com.losandroides.learn.domain.toDomain
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class DefaultLocalItemDatasource @Inject constructor(
+    database: ItemDatabase
+) : LocalItemDatasource {
+
+    private val dao = database.itemDatabaseModelDao()
+
+    override suspend fun isEmpty(): Boolean =
+        withContext(Dispatchers.IO) { dao.itemsCount() <= 0 }
```

Comment:
The dispatchers could be provided by the DI.

Reply:
True!